modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1905,10 +1905,13 @@ public Parameter getParameter(ObjectNode obj, String location, ParseResult resul
 				parameter = new Parameter();
 				String mungedRef = mungedRef(ref.textValue());
 				if (mungedRef != null) {
-					parameter.set$ref(mungedRef);
-				} else {
-					parameter.set$ref(ref.textValue());
-				}
+                    parameter.set$ref(mungedRef);
+                }else {
+                    parameter.set$ref(ref.textValue());
+                }
+                if(ref.textValue().startsWith("#/components") && !(ref.textValue().startsWith("#/components/parameters")||ref.textValue().startsWith("#/components/headers"))) {
+                    result.warning(location, "$ref is not pointing to #/components/parameters");
```

Comment:
possibly change to something like `$ref target #/components/headers/whatever is not of expected type 'header'`. @ponelat possibly has some even better wording